test/automation/src/sql/connectionDialog.ts

```diff
@@ -31,7 +31,9 @@ export class ConnectionDialog extends Dialog {
 	private static readonly CONNECT_BUTTON_SELECTOR = '.modal .modal-footer a[aria-label="Connect"]:not(.disabled)';
 	async connect(): Promise<void> {
 		await this.code.waitAndClick(ConnectionDialog.CONNECT_BUTTON_SELECTOR);
-
-		return this.waitForDialogGone();
+		// Try pressing the connect button again if the dialog does not go away
```

Comment:
Hmmm, but why would the first click not do anything? We might as well do a timeout or something here - but those are both just trying to hide some other issue.

As Alan suggested is there some other thing we can do to determine why the "click" didn't seem to take? Maybe this is a timing issue with the click handler for the button or something weird? Would be nice to add in some logging in the ADS code to indicate that we received the click to help figure out what's going on here.

Reply:
I'll add some logging in connectionDialogWidget.ts and connectionDialogService.ts to see if connect is actually being called.